### PR TITLE
[Structural] fix error in debug

### PR DESCRIPTION
**Pull request by philbucher**
otherwise throws if uninitialized

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_elements/nodal_concentrated_element.cpp [hunk 1]
```diff
@@ -422,9 +422,11 @@ void NodalConcentratedElement::CalculateDampingMatrix(
             rCurrentProcessInfo,
             system_size);
     } else {
-        const array_1d<double, 3 >& nodal_damping_ratio = this->GetValue(NODAL_DAMPING_RATIO);
-        for ( unsigned int j = 0; j < dimension; ++j )
+        const auto& rconst_this = *this;
+        const array_1d<double, 3 >& nodal_damping_ratio = rconst_this.GetValue(NODAL_DAMPING_RATIO);
+        for ( unsigned int j = 0; j < dimension; ++j ) {
```
Comment (tteschemacher) [inline applications/StructuralMechanicsApplication/custom_elements/nodal_concentrated_element.cpp:427]: You can make this SizeType.

**Review by tteschemacher — COMMENTED** [review]
That looks obvious to me. But are you sure that this operation is cheaper than just obtaining it non const? Shouldn't that have the same result?

File: applications/StructuralMechanicsApplication/custom_elements/nodal_concentrated_element.cpp (hunk 1 above)
Comment (philbucher) [reply applications/StructuralMechanicsApplication/custom_elements/nodal_concentrated_element.cpp:427]: since this is a cosmetic change I would not do it in this PR
Also then the rest of the element should be changed

**Comment by philbucher** [comment]
@tteschemacher (and possibly others reading this) I cannot get it from the non-const version! The reason is in the implementation of the [DataValueContainer](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/containers/data_value_container.h#L208-L241). 
If I get it non-const, then it will allocate the variable if it does not exist yet (which is the case here). This throws if run in omp. Whether this is a bug or a feature is a separate discussion ;)
Long story short, I have to use the const version, since this one obviously does not allocate it.

Hence one has to be careful, that is one of the reasons why I am often suggesting to make functions const.

**Review by tteschemacher — APPROVED** [review]
Good to know. Thanks!
